lang/tests/src/test/scala/com/wavesplatform/lang/evaluator/BytesDropTakeTest.scala

```diff
@@ -0,0 +1,76 @@
+    eval(s""" take($b, $min) """, V1, V5) shouldBe bytes()
+    eval(s""" take($b, $min) """)(V6) shouldBe Left(s"Unexpected negative number = $min passed to take()")
+  }
+
+  property("takeRight") {
+    eval(s""" takeRight($b, 0) """) shouldBe bytes()
+    eval(s""" takeRight($b, 2) """) shouldBe bytes(4, 5)
+    eval(s""" takeRight($b, 100) """) shouldBe bytes(1, 2, 3, 4, 5)
+
+    eval(s""" takeRight($b, $max) """, V1, V5) shouldBe bytes()
+    eval(s""" takeRight($b, $max) """)(V6) shouldBe Left(s"Number = $max passed to takeRight() exceeds ByteVector limit = $limit")
+    eval(s""" takeRight($b, ${limit + 1}) """)(V6) shouldBe Left(s"Number = ${limit + 1} passed to takeRight() exceeds ByteVector limit = $limit")
+
+    eval(s""" takeRight($b, -1) """, V1, V5) shouldBe bytes()
+    eval(s""" takeRight($b, -1) """)(V6) shouldBe Left("Unexpected negative number = -1 passed to takeRight()")
+    eval(s""" takeRight($b, $min) """, V1, V5) shouldBe Left("long overflow")
+    eval(s""" takeRight($b, $min) """)(V6) shouldBe Left(s"Unexpected negative number = $min passed to takeRight()")
+  }
+
+  property("drop") {
+    eval(s""" drop($b, 0) """) shouldBe bytes(1, 2, 3, 4, 5)
+    eval(s""" drop($b, 2) """) shouldBe bytes(3, 4, 5)
+    eval(s""" drop($b, 100) """) shouldBe bytes()
+
+    eval(s""" drop($b, $max) """, V1, V5) shouldBe bytes(1, 2, 3, 4, 5)
+    eval(s""" drop($b, $max) """)(V6) shouldBe Left(s"Number = $max passed to drop() exceeds ByteVector limit = $limit")
+    eval(s""" drop($b, ${limit + 1}) """)(V6) shouldBe Left(s"Number = ${limit + 1} passed to drop() exceeds ByteVector limit = $limit")
+
+    eval(s""" drop($b, -1) """, V1, V5) shouldBe bytes(1, 2, 3, 4, 5)
+    eval(s""" drop($b, -1) """)(V6) shouldBe Left("Unexpected negative number = -1 passed to drop()")
+    eval(s""" drop($b, $min) """, V1, V5) shouldBe bytes(1, 2, 3, 4, 5)
+    eval(s""" drop($b, $min) """)(V6) shouldBe Left(s"Unexpected negative number = $min passed to drop()")
+  }
+
+  property("dropRight") {
+    eval(s""" dropRight($b, 0) """) shouldBe bytes(1, 2, 3, 4, 5)
+    eval(s""" dropRight($b, 2) """) shouldBe bytes(1, 2, 3)
+    eval(s""" dropRight($b, 100) """) shouldBe bytes()
+
+    eval(s""" dropRight($b, $max) """, V1, V5) shouldBe bytes(1, 2, 3, 4, 5)
+    eval(s""" dropRight($b, $max) """)(V6) shouldBe Left(s"Number = $max passed to dropRight() exceeds ByteVector limit = $limit")
+    eval(s""" dropRight($b, ${limit + 1}) """)(V6) shouldBe Left(s"Number = ${limit + 1} passed to dropRight() exceeds ByteVector limit = $limit")
+
+    eval(s""" dropRight($b, -1) """, V1, V5) shouldBe bytes(1, 2, 3, 4, 5)
+    eval(s""" dropRight($b, -1) """)(V6) shouldBe Left("Unexpected negative number = -1 passed to dropRight()")
+    eval(s""" dropRight($b, $min) """, V1, V5) should produce("long overflow")
+    eval(s""" dropRight($b, $min) """)(V6) shouldBe Left(s"Unexpected negative number = $min passed to dropRight()")
+  }
```

Comment:
Add tests for each of the functions, using byte vector with size = limit
A reference for what i mean from `StringFunctionsTest`:
`eval(s"""take("$maxString", $limit) """) shouldBe CONST_STRING(maxString)`

Reply:
Done